Sources/AdjustRemoteCommand.swift

```diff
@@ -144,11 +144,13 @@ public class AdjustRemoteCommand: RemoteCommand {
             case .gdprForgetMe:
                 adjustInstance.gdprForgetMe()
             case .setThirdPartySharing:
-                guard let enabled = payload[AdjustConstants.Keys.enabled] as? Bool else {
-                    log("\(AdjustConstants.Keys.enabled) required")
+                let enabled = payload[AdjustConstants.Keys.enabled] as? Bool
+                let granularOptions = payload[AdjustConstants.Keys.thirdPartySharingOptions] as? [String: [String: String]]
+                if (enabled == nil && granularOptions == nil) {
+                    log("\(AdjustConstants.Keys.enabled) or \(AdjustConstants.Keys.thirdPartySharingOptions) required")
                     return
                 }
```

Comment:
Instead of this we could use the swift way to do the checks:

```swift
guard enabled != nil || granularOptions != nil else {
    log("\(AdjustConstants.Keys.enabled) or \(AdjustConstants.Keys.thirdPartySharingOptions) required")
    return
}
```